source/patterns/@aws-solutions-constructs/aws-wafwebacl-apigateway/README.md

```diff
@@ -0,0 +1,104 @@
+
+
+| **Language**     | **Package**        |
+|:-------------|-----------------|
+|![Python Logo](https://docs.aws.amazon.com/cdk/api/latest/img/python32.png) Python|`aws_solutions_constructs.aws_wafwebacl_apigateway`|
+|![Typescript Logo](https://docs.aws.amazon.com/cdk/api/latest/img/typescript32.png) Typescript|`@aws-solutions-constructs/aws-wafwebacl-apigateway`|
+|![Java Logo](https://docs.aws.amazon.com/cdk/api/latest/img/java32.png) Java|`software.amazon.awsconstructs.services.wafwebaclapigateway`|
+
+## Overview
+This AWS Solutions Construct implements an AWS WAF connected to Amazon API Gateway REST API.
+
+Here is a minimal deployable pattern definition in Typescript:
+
+``` typescript
+import * as api from '@aws-cdk/aws-apigateway';
+import * as lambda from "@aws-cdk/aws-lambda";
+import { WafWebACLToApiGatewayProps, WafWebACLToApiGateway } from "@aws-solutions-constructs/aws-wafwebacl-apigateway";
+
+const lambdaProps: lambda.FunctionProps = {
+    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    runtime: lambda.Runtime.NODEJS_12_X,
+    handler: 'index.handler'
+};
+
+const lambdafunction = new lambda.Function(this, 'LambdaFunction', lambdaProps);
+const apiGateway = new api.RestApi(this, 'test-api');
+const tasks = apiGateway.root.addResource('tasks');
+
+apiGateway.addMethod('GET', new api.LambdaIntegration(lambdaFunction));
+
+const apigatewayInterface = api.RestApi.fromRestApi(this, 'apigateway-interface', apiGateway.restApiId);
+
+new WafWebACLToApiGateway(this, 'test-wafwebacl-apigateway', {
+    existingApiGateway: apigatewayInterface
+});
+
+```
+
+## Initializer
+
+``` text
+new WafWebACLToApiGateway(scope: Construct, id: string, props: WafWebACLToApiGatewayProps);
+```
+
+_Parameters_
+
+* scope [`Construct`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Construct.html)
+* id `string`
+* props [`WafWebACLToApiGatewayProps`](#pattern-construct-props)
+
+## Pattern Construct Props
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+|existingApiGateway|[`api.IRestApi`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-apigateway.IRestApi.html)|The existing API Gateway instance that will be fronted with the WAF.|
+|webACLProps?|[`waf.CfnWebACLProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-waf.CfnWebACLProps.html)|Optional user-provided props to override the default props for the AWS WAF access control list.|
+
+## Pattern Properties
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+|webACL|[`waf.CfnWebACL`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-waf.CfnWebACL.html)|Returns an instance of the waf.CfnWebACL created by the construct.|
+|apiGateway|[`api.RestApi`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-apigateway.RestApi.html)|	Returns an instance of the API Gateway REST API created by the pattern. |
```

Comment:
Expose IRestApi, not RestApi. When they send us an IRestApi we have no way to convert it to a RestApi.